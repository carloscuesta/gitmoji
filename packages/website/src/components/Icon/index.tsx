export { IconDefinitions } from './definitions'
import styles from './styles.module.css'

type Props = { name: string }

const Icon = (props: Props) => (
  <svg className={`${styles.icon} icon-${props.name}`}>
    <use xlinkHref={`#icon-${props.name}`} />
  </svg>
)

export default Icon
