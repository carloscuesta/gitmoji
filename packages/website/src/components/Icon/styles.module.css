.icon {
  width: 1em;
  height: 1em;
  margin-right: 0.25em;
}

.icon:global(.icon-heart) {
  margin: 0;
}
