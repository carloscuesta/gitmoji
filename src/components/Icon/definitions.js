// @flow
import React, { type Element } from 'react'

export const IconDefinitions = (): Element<'svg'> => (
  <svg
    style={{ position: 'absolute', width: 0, height: 0 }}
    width={0}
    height={0}
    version="1.1"
    xmlns="http://www.w3.org/2000/svg"
    xmlnsXlink="http://www.w3.org/1999/xlink"
  >
    <defs>
      <symbol id="icon-heart" viewBox="0 0 64 64">
        <title>heart</title>
        <path
          className="heart"
          d="m61.1 18.2c-6.4-17-27.2-9.4-29.1-.9-2.6-9-22.9-15.7-29.1.9-6.9 18.5 26.7 35.1 29.1 37.8 2.4-2.2 36-19.6 29.1-37.8"
          fill="#ff5a79"
        />
      </symbol>
      <symbol id="icon-star" viewBox="0 0 64 64">
        <title>star</title>
        <path
          className="twitter"
          d="M62,25.2H39.1L32,3l-7.1,22.2H2l18.5,13.7l-7,22.1L32,47.3L50.5,61l-7.1-22.2L62,25.2z"
          fill="#FFDD67"
        />
      </symbol>
      <symbol id="icon-twitter" viewBox="0 0 64 64">
        <title>twitter</title>
        <g fill="#42ade2">
          <path d="m59.8 24.3c0 0 1.1-6.2-3.5-3.4 0 0-.4-6.3-4.3-1.9 0 0-2.1-3.9-4.4-.3-3.1 4.8-5.2 12.4-3.2 25l3.8-2.5c2.7-7.9 12.4-8.8 13.7-13.1.9-3-2.1-3.8-2.1-3.8" />
          <path d="m22.1 17.6l-9.9 3.6c2.2-12 16.6-11.2 16.6-11.2s-6.8 3.2-6.7 7.6" />
          <path d="m23.7 19.8l-10.5 1.4c4.8-11.2 18.7-7.3 18.7-7.3s-7.3 1.6-8.2 5.9" />
        </g>
        <g fill="#ffd93b">
          <path d="m2 29l5.4-1.4v3.6c0-.1-3.3-.6-5.4-2.2" />
          <path d="M7.4,27.5L2,24.8c3.6-2.8,7.7-1.9,7.7-1.9L7.4,27.5z" />
        </g>
        <g fill="#e08828">
          <path d="m33.8 53h-2.1v7.9c-.3.1-2.1-.1-2.9-.1-1.8 0-3.3 1.3-3.3 1.3h8.3v-9.1" />
          <path d="m25 53h-2.1v7.9c-.3.1-2.1-.1-2.9-.1-1.8 0-3.3 1.3-3.3 1.3h8.3v-9.1" />
          <path
            d="m54 36.2c3.9 0-4.1 17.5-23.3 17.5-13 0-23.9-5.2-23.9-21.5 0-10.1 6.4-18.3 19.5-15 13.3 3.5 6.5 19 27.7 19"
            fill="#42ade2"
          />
          <path
            d="m37.6 51.7c-15.6 0-14-12-27.9-11.2 5.1 15.8 27.9 11.2 27.9 11.2"
            fill="#fff"
          />
          <path
            d="m39.1 29.2c-10-9.8-20.2 6.2-7.9 12.6 12.1 6.2 20.4-4.8 20.4-4.8s-6.1-1.5-12.5-7.8"
            fill="#297b9d"
          />
        </g>
        <circle cx="15.1" cy="24.9" r="2.5" fill="#3e4347" />
      </symbol>
    </defs>
  </svg>
)
