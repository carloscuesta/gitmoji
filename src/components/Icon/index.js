// @flow
import React, { type Element } from 'react'

export { IconDefinitions } from './definitions'

type Props = { name: string }

const Icon = (props: Props): Element<'svg'> => (
  <svg className={`icon icon-${props.name}`}>
    <use xlinkHref={`#icon-${props.name}`} />
  </svg>
)

export default Icon
