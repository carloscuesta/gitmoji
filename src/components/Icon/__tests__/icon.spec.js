import renderer from 'react-test-renderer'

import Icon, { IconDefinitions } from '../index'
import * as stubs from './stubs'

describe('Icon', () => {
  it('should render the component', () => {
    const wrapper = renderer.create(<Icon {...stubs.props} />)
    expect(wrapper).toMatchSnapshot()
  })

  it('should match Definitions', () => {
    const wrapper = renderer.create(<IconDefinitions />)
    expect(wrapper).toMatchSnapshot()
  })
})
