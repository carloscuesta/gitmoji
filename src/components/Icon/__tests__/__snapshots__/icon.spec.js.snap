// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon should match Definitions 1`] = `
<svg
  height={0}
  style={
    Object {
      "height": 0,
      "position": "absolute",
      "width": 0,
    }
  }
  version="1.1"
  width={0}
  xmlns="http://www.w3.org/2000/svg"
  xmlnsXlink="http://www.w3.org/1999/xlink"
>
  <defs>
    <symbol
      id="icon-heart"
      viewBox="0 0 64 64"
    >
      <title>
        heart
      </title>
      <path
        className="heart"
        d="m61.1 18.2c-6.4-17-27.2-9.4-29.1-.9-2.6-9-22.9-15.7-29.1.9-6.9 18.5 26.7 35.1 29.1 37.8 2.4-2.2 36-19.6 29.1-37.8"
        fill="#ff5a79"
      />
    </symbol>
    <symbol
      id="icon-star"
      viewBox="0 0 64 64"
    >
      <title>
        star
      </title>
      <path
        className="twitter"
        d="M62,25.2H39.1L32,3l-7.1,22.2H2l18.5,13.7l-7,22.1L32,47.3L50.5,61l-7.1-22.2L62,25.2z"
        fill="#FFDD67"
      />
    </symbol>
    <symbol
      id="icon-twitter"
      viewBox="0 0 64 64"
    >
      <title>
        twitter
      </title>
      <g
        fill="#42ade2"
      >
        <path
          d="m59.8 24.3c0 0 1.1-6.2-3.5-3.4 0 0-.4-6.3-4.3-1.9 0 0-2.1-3.9-4.4-.3-3.1 4.8-5.2 12.4-3.2 25l3.8-2.5c2.7-7.9 12.4-8.8 13.7-13.1.9-3-2.1-3.8-2.1-3.8"
        />
        <path
          d="m22.1 17.6l-9.9 3.6c2.2-12 16.6-11.2 16.6-11.2s-6.8 3.2-6.7 7.6"
        />
        <path
          d="m23.7 19.8l-10.5 1.4c4.8-11.2 18.7-7.3 18.7-7.3s-7.3 1.6-8.2 5.9"
        />
      </g>
      <g
        fill="#ffd93b"
      >
        <path
          d="m2 29l5.4-1.4v3.6c0-.1-3.3-.6-5.4-2.2"
        />
        <path
          d="M7.4,27.5L2,24.8c3.6-2.8,7.7-1.9,7.7-1.9L7.4,27.5z"
        />
      </g>
      <g
        fill="#e08828"
      >
        <path
          d="m33.8 53h-2.1v7.9c-.3.1-2.1-.1-2.9-.1-1.8 0-3.3 1.3-3.3 1.3h8.3v-9.1"
        />
        <path
          d="m25 53h-2.1v7.9c-.3.1-2.1-.1-2.9-.1-1.8 0-3.3 1.3-3.3 1.3h8.3v-9.1"
        />
        <path
          d="m54 36.2c3.9 0-4.1 17.5-23.3 17.5-13 0-23.9-5.2-23.9-21.5 0-10.1 6.4-18.3 19.5-15 13.3 3.5 6.5 19 27.7 19"
          fill="#42ade2"
        />
        <path
          d="m37.6 51.7c-15.6 0-14-12-27.9-11.2 5.1 15.8 27.9 11.2 27.9 11.2"
          fill="#fff"
        />
        <path
          d="m39.1 29.2c-10-9.8-20.2 6.2-7.9 12.6 12.1 6.2 20.4-4.8 20.4-4.8s-6.1-1.5-12.5-7.8"
          fill="#297b9d"
        />
      </g>
      <circle
        cx="15.1"
        cy="24.9"
        fill="#3e4347"
        r="2.5"
      />
    </symbol>
    <symbol
      id="icon-grid"
      viewBox="0 0 359 359"
      x="0px"
      y="0px"
    >
      <title>
        grid
      </title>
      <g>
        <g>
          <path
            d="M96,0H13C7.5,0,3,4.5,3,10v83c0,5.5,4.5,10,10,10h83c5.5,0,10-4.5,10-10V10C106,4.5,101.5,0,96,0z M87,84H23V20h64V84z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M221,0h-83c-5.5,0-10,4.5-10,10v83c0,5.5,4.5,10,10,10h83c5.5,0,10-4.5,10-10V10C231,4.5,226.5,0,221,0z M212,84h-64V20
h64V84z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M96,126H13c-5.5,0-10,4.5-10,10v83c0,5.5,4.5,10,10,10h83c5.5,0,10-4.5,10-10v-83C106,130.5,101.5,126,96,126z M87,209H23
v-63h64V209z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M221,126h-83c-5.5,0-10,4.5-10,10v83c0,5.5,4.5,10,10,10h83c5.5,0,10-4.5,10-10v-83C231,130.5,226.5,126,221,126z
M212,209h-64v-63h64V209z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M346,0h-83c-5.5,0-10,4.5-10,10v83c0,5.5,4.5,10,10,10h83c5.5,0,10-4.5,10-10V10C356,4.5,351.5,0,346,0z M337,84h-64V20
h64V84z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M346,126h-83c-5.5,0-10,4.5-10,10v83c0,5.5,4.5,10,10,10h83c5.5,0,10-4.5,10-10v-83C356,130.5,351.5,126,346,126z
M337,209h-64v-63h64V209z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M96,256H13c-5.5,0-10,4.5-10,10v83c0,5.5,4.5,10,10,10h83c5.5,0,10-4.5,10-10v-83C106,260.5,101.5,256,96,256z M87,340H23
v-64h64V340z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M221,256h-83c-5.5,0-10,4.5-10,10v83c0,5.5,4.5,10,10,10h83c5.5,0,10-4.5,10-10v-83C231,260.5,226.5,256,221,256z
M212,340h-64v-64h64V340z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M346,256h-83c-5.5,0-10,4.5-10,10v83c0,5.5,4.5,10,10,10h83c5.5,0,10-4.5,10-10v-83C356,260.5,351.5,256,346,256z
M337,340h-64v-64h64V340z"
          />
        </g>
      </g>
    </symbol>
    <symbol
      id="icon-list"
      viewBox="0 0 343.5 343.5"
      x="0px"
      y="0px"
    >
      <title>
        list
      </title>
      <g>
        <g>
          <path
            d="M322.05,161.8h-182.6c-5.5,0-10,4.5-10,10s4.5,10,10,10h182.6c5.5,0,10-4.5,10-10C332.05,166.3,327.65,161.8,322.05,161.8
 z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M57.95,125.3c-25.7,0-46.5,20.8-46.5,46.5s20.8,46.5,46.5,46.5s46.5-20.8,46.5-46.5S83.65,125.3,57.95,125.3z
 M57.95,198.3c-14.7,0-26.5-11.9-26.5-26.5c0-14.7,11.9-26.5,26.5-26.5c14.6,0,26.5,11.9,26.5,26.5S72.55,198.3,57.95,198.3z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M322.05,36.8h-182.6c-5.5,0-10,4.5-10,10s4.5,10,10,10h182.6c5.5,0,10-4.5,10-10C332.05,41.3,327.65,36.8,322.05,36.8z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M57.95,0c-25.7,0-46.5,20.8-46.5,46.5c0,25.7,20.8,46.5,46.5,46.5s46.5-20.8,46.5-46.5C104.45,20.9,83.65,0.1,57.95,0z
 M57.95,73.1c-14.7,0-26.5-11.9-26.5-26.5c0-14.6,11.9-26.5,26.5-26.5c14.7,0,26.5,11.9,26.5,26.5
 C84.45,61.2,72.55,73.1,57.95,73.1z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M322.05,286.8h-182.6c-5.5,0-10,4.5-10,10s4.5,10,10,10h182.6c5.5,0,10-4.5,10-10S327.65,286.8,322.05,286.8z"
          />
        </g>
      </g>
      <g>
        <g>
          <path
            d="M57.95,250.5c-25.7,0-46.5,20.8-46.5,46.5c0,25.7,20.8,46.5,46.5,46.5s46.5-20.8,46.5-46.5
 C104.45,271.4,83.65,250.5,57.95,250.5z M57.95,323.6c-14.7,0-26.5-11.9-26.5-26.5c0-14.7,11.9-26.5,26.5-26.5
 c14.7,0,26.5,11.9,26.5,26.5S72.55,323.6,57.95,323.6z"
          />
        </g>
      </g>
    </symbol>
  </defs>
</svg>
`;

exports[`Icon should render the component 1`] = `
<svg
  className="icon icon-star"
>
  <use
    xlinkHref="#icon-star"
  />
</svg>
`;
